lib/src/r_tree/r_tree.dart

```diff
@@ -52,7 +52,7 @@ class RTree<E> {
   //  Note: Rectangles that share only a border are not considered to overlap
   Iterable<RTreeDatum<E>> search(Rectangle searchRect) {
     if (_root is LeafNode) {
-      return _root.search(searchRect).toList();
+      return _root.search(searchRect).toList() as List<RTreeDatum<E>>;
```

Comment:
I don't think this cast should be done, LeafNode seems like it should probably be LeafNode<E> so that search can return the correct type?